src/qibo/models/circuit.py

```diff
@@ -1012,6 +1024,41 @@ def __call__(self, initial_state=None, nshots=None):
         """Equivalent to ``circuit.execute``."""
         return self.execute(initial_state=initial_state, nshots=nshots)
 
+    @property
+    def raw(self) -> dict:
+        """Serialize to dictionary.
+
+        This is a thin wrapper over :meth:`Gate.raw`.
+        """
+        return {
+            "queue": [gate.raw for gate in self.queue],
+            "nqubits": self.nqubits,
+            "density_matrix": self.density_matrix,
+            "qibo_version": qibo.__version__,
+        }
+
+    @classmethod
+    def from_dict(cls, raw):
+        """Load from serialization.
+
+        Essentially the counter-part of :meth:`raw`.
+        """
+        circ = cls(raw["nqubits"], density_matrix=raw["density_matrix"])
+
+        for gate in raw["queue"]:
+            circ.add(Gate.load(gate))
+
+        return circ
+
+    def dump_json(self, stream):
+        """Dump json representation to the given stream."""
+        json.dump(self.raw, stream)
+
+    @classmethod
+    def load_json(cls, stream):
+        """Dump json representation to the given stream."""
+        return cls.load(json.load(stream))
```

Comment:
This `cls.load` does seem to exists in the code.

Reply:
Yes, it was a leftover. I received the Pylint notification.